src/main/java/org/isf/visits/service/VisitsIoOperations.java

```diff
@@ -52,7 +53,26 @@
 				new ArrayList<>(repository.findAllByPatient_CodeOrderByPatient_CodeAscDateAsc(patID)) :
 				new ArrayList<>(repository.findAllByOrderByPatient_CodeAscDateAsc());
 	}
+	
+	/**
+	 * Returns the list of all {@link Visit}s related to a patID in OPD (Ward is {@code null}).
+	 *
+	 * @param patID - the {@link Patient} ID. If <code>0</code> return the list of all OPD {@link Visit}s
+	 * @return the list of {@link Visit}s
+	 * @throws OHServiceException
+	 */
+	public ArrayList<Visit> getVisitsOPD(Integer patID) throws OHServiceException {
+		return patID != 0 ?
+				new ArrayList<>(repository.findAllByWardIsNullAndPatient_CodeOrderByPatient_CodeAscDateAsc(patID)) :
+				new ArrayList<>(repository.findAllByWardIsNullOrderByPatient_CodeAscDateAsc());
```

Comment:
And if you make the change noted above then you don't need `new ArrayList<>` because the repository method returns a `List`.

Reply:
correct! thanks!